cpp/include/cudf/detail/sequence.hpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/filling.hpp>
+#include <cudf/detail/sequence.hpp>
+#include <cudf/types.hpp>
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+
+/**
+ * @copydoc cudf::experimental::sequence
+ *
+ * @param stream CUDA stream to run this function
+ **/
+std::unique_ptr<column> sequence(size_type size, scalar const& init, scalar const& step,
```

Comment:
What type does this API return? Int32, Int64? Should we add a type parameter and dispatch on it?

Reply:
If we're going to go that far, we could make a templated version that takes a straight constant that defaults to 1. Internally, it'll need a seperate functor so may as well expose it generally. Something like?

```
template<typename T>
std::unique_ptr<column> sequence(size_type size, scalar const& init, T step = 1,
                                 rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
```

Reply:
Although I guess if we did that, it would be on the user to get the template type to match with the scalar, which is kind of gross.

Reply:
Yeah, we can't really do templates in top-level APIs.

Reply:
There's no need for a template version in the first place: it's just an iterator!

Reply:
The type of the counting iterator should still match the type of the init.

Reply:
Yeah, this is all moot because we can't have templates in the public API. My point was just that if we were a template library, we probably wouldn't need `sequence` at all.